src/wc_avx2.c

```diff
@@ -0,0 +1,115 @@
+/* wc - print the number of lines, words, and bytes in files
+   Copyright (C) 1985-2021 Free Software Foundation, Inc.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.  */
+
+#include <config.h>
+
+#include "system.h"
+#include "die.h"
+#include "safe-read.h"
+
+#include <x86intrin.h>
+
+/* This must be below 16 KB (16384) or else the accumulators can theoretically overflow,
+   producing wrong result. This is 2*32 bytes below, so there is no single bytes in the
+   optimal case. */
+#define BUFSIZE (16320)
+
+bool wc_lines_avx2(const char *file, int fd, uintmax_t *lines_out, uintmax_t *bytes_out);
+
+bool wc_lines_avx2(const char *file, int fd, uintmax_t *lines_out, uintmax_t *bytes_out)
+{
+  __m256i accumulator;
+  __m256i accumulator2;
+  __m256i zeroes;
+  __m256i endlines;
+  __m256i avx_buf[BUFSIZE / sizeof(__m256i)];
```

Comment:
This should specify an explicit buffer size, so that the 8-bit counters cannot overflow.

Reply:
Hm.. can you explain more what you propose? Instead of using BUFSIZE (defined above to 16320), instead use 16320 directly? 
The size 16320 is calculated so that the counters in the accumulator can max become 255 and not wrap.

Reply:
Oh, I confused it with `BUFSIZ`. I don't think there is a need for a macro. And page alignment may confer additional performance benefits (8192 bytes aligned to a 4K page).